### PR TITLE
Events

### PR DESCRIPTION
**Pull request by darmhoo**
View all event
Create Event
View one event

### CODE REVIEW

**Review by Golden-Ogbeka — CHANGES_REQUESTED** [review]
- There are some traces of devotional inside the event controllers
- The event date isn't unique. One date can have multiple events. The check for a unique date in add event isn't required 
- The extra fields for date or number limit in events isn't required from the frontend. The validation makes them required by default. They should be added only if the frontend sends them not compulsorily

**Comment by Golden-Ogbeka** [comment]
Does Tern JS provide the language and coding support of Typescript for Javascript files?

**Comment by darmhoo** [comment]
I don't understand the third comment. I didn't include the validation in the route file because they are not compulsory

File: api/v1/routes/event.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            .exists()
+            .bail()
+            .custom((value) => isAdmin(value)),
+        body('date', 'Date is required').trim().exists().notEmpty().withMessage("Date cannot be empty").isISO8601().toDate().withMessage("Enter a valid date"),
+
+        body('name', 'Name is required').trim().exists().notEmpty().withMessage("Nmae cannot be empty"),
+        body('theme', 'Theme is required').trim().exists().notEmpty().withMessage("Theme cannot be empty"),
+        body('mainText', 'Main Text is required').trim().exists().notEmpty().withMessage("Main Text cannot be empty"),
+        body('time', 'Time is required').trim().exists().notEmpty().withMessage("Time cannot be empty"),
+        body('allowRegistration', 'Allow Registration is required').trim().exists().isBoolean().withMessage("Allow Registration must be boolean"),
```
Comment (darmhoo) [inline api/v1/routes/event.ts:43]: Do you mean this part

Comment (darmhoo) [reply api/v1/routes/event.ts:43]: @Golden-Ogbeka

Comment (Golden-Ogbeka) [reply api/v1/routes/event.ts:43]: Yeah, the .exists means it is required

Comment (Golden-Ogbeka) [reply api/v1/routes/event.ts:43]: @darmhoo if you try testing the routes without those fields, it shouldnt go through

Comment (darmhoo) [reply api/v1/routes/event.ts:43]: Ok. i will fix now

**Comment by Golden-Ogbeka** [comment]
@darmhoo you pushed for theme and main text but we'll proceed with this so testing can begin. Later on, we'll change it for fields that arent compulsort

**Review by Golden-Ogbeka — APPROVED** [review]

**Pull request merged**